lib/ui/painting/gradient.cc

```diff
@@ -45,10 +45,14 @@ void CanvasGradient::initLinear(const tonic::Float32List& end_points,
 
   SkPoint p0 = SkPoint::Make(end_points[0], end_points[1]);
   SkPoint p1 = SkPoint::Make(end_points[2], end_points[3]);
-  const DlColor* colors_array = reinterpret_cast<const DlColor*>(colors.data());
+  std::vector<DlColor> dl_colors;
+  dl_colors.reserve(colors.num_elements());
+  for (int i = 0; i < colors.num_elements(); ++i) {
+    dl_colors.emplace_back(DlColor(colors[i]));
```

Comment:
I used emplace since the argument is an rvalue.  I think this should be as fast as push_back even though DlColor has no pointers.  I thought I remember something in the c++ style guide about preferring push_back in certain cases but it isn't in there anymore.

Reply:
I guess its an rvalue because you constructed a DlColor here? will that really initialize DlColor in place? gosh c++ is confusing sometimes.

Reply:
> I guess its an rvalue because you constructed a DlColor here?
Yep.

> will that really initialize DlColor in place?
It's supposed to.  I think with the return value optimization `push_back` should be just as fast.  It's just an easier rule to follow I think if rvalues use the rvalue methods.

Reply:
Can you do emplace_back(colors[i])? Or is that disabled because of the explicit constructor?

Reply:
> Can you do emplace_back(colors[i])? Or is that disabled because of the explicit constructor?

yea you need an explicit constructor

Reply:
I figured that since it was `emplace_back` it would qualify as an explicit call to a constructor.